lib/dry/struct.rb

```diff
@@ -167,12 +170,11 @@ def new(changeset)
     end
     alias_method :__new__, :new
 
-    # @return[Hash{Symbol => Object}]
-    # @api private
-    def __attributes__
-      self.class.attribute_names.each_with_object({}) do |key, h|
-        h[key] = instance_variable_get(:"@#{ key }")
-      end
+    # @return [String]
+    def inspect
+      klass = self.class
+      attrs = klass.attribute_names.map { |key| " #{key}=#{@attributes[key].inspect}" }.join
```

Comment:
I don't use `@attributes.map` because it doesn't preserve the order of attributes in the schema. This is an accident change in master of `dry-types` but I don't think it's important